tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/NotEqualOperationTest.java

```diff
@@ -130,9 +130,9 @@ public void testStringNotEqualExpr() {
 
     @Test(description = "Test not equal expression statement with errors")
     public void testNotEqualStmtNegativeCases() {
-        Assert.assertEquals(resultNegative.getErrorCount(), 3);
+        Assert.assertEquals(resultNegative.getErrorCount(), 1);
         BAssertUtil.validateError(resultNegative, 0, "operator '==' not defined for 'int' and 'string'", 4, 12);
-        BAssertUtil.validateError(resultNegative, 1, "operator '==' not defined for 'json' and 'json'", 13, 12);
-        BAssertUtil.validateError(resultNegative, 2, "operator '!=' not defined for 'json' and 'json'", 22, 12);
+        //BAssertUtil.validateError(resultNegative, 1, "operator '==' not defined for 'json' and 'json'", 13, 12);
+        //BAssertUtil.validateError(resultNegative, 2, "operator '!=' not defined for 'json' and 'json'", 22, 12);
```

Comment:
Current impl check for reference equality. This PR do not change the behavior. However these negative cases were removed since we allow these types of equalities at compile time.